### PR TITLE
fix: handle correct hostnam

### PR DESCRIPTION
**Pull request by tanguyantoine**
## 🐻 What your PR is doing?

-  handle missing hostname option

http args allow either host or hostname as argument

## 📦 Package concerned
- @bearer/x-ray

## ✅ Checklist

- [x] Tests were added (if necessary)
- [x] I used conventional commits

### CODE REVIEW

File: packages/x-ray/src/http-overrides.ts [hunk 1]
```diff
@@ -54,13 +54,15 @@ export const overrideRequestMethod = (
...
+      // http accepts either host or hostname
+      parsedUrl.protocol = parsedUrl.protocol || 'http'
+      parsedUrl.hostname = parsedUrl.hostname || parsedUrl.host
     }
-    const { port, path, host, protocol, auth, hostname, hash, search, query, pathname, href } = parsedUrl
+    const { port, path, protocol, auth, hostname, hash, search, query, pathname, href } = parsedUrl
     info.request = {
       ...info.request,
       port,
       path,
```
Comment (numidiasoft) [inline packages/x-ray/src/http-overrides.ts]: What is the diff between the path and the pathname ?

File: packages/x-ray/src/http-overrides.ts [hunk 2]
```diff
@@ -54,13 +54,15 @@ export const overrideRequestMethod = (
       parsedUrl = urlParser.parse(url, true)
     } else {
       parsedUrl = url
+      // http accepts either host or hostname
+      parsedUrl.protocol = parsedUrl.protocol || 'http'
+      parsedUrl.hostname = parsedUrl.hostname || parsedUrl.host
     }
-    const { port, path, host, protocol, auth, hostname, hash, search, query, pathname, href } = parsedUrl
```
Comment (cfabianski) [inline packages/x-ray/src/http-overrides.ts]: let's keep `host` as well maybe?

Comment (tanguyantoine) [reply packages/x-ray/src/http-overrides.ts]: `host`: The full lowercased host portion of the URL, including port information.
`hostname:` Just the lowercased hostname portion of the host.

same as path and pathname

File: packages/x-ray/src/http-overrides.ts (hunk 1 above)
Comment (tanguyantoine) [reply packages/x-ray/src/http-overrides.ts]: https://github.com/defunctzombie/node-url#api:
`pathname`: The path section of the URL, that comes after the host and before the query, including the initial slash if present.
`path`: Concatenation of pathname and search.

**Pull request merged**